File: check.py
from os import environ
from google.cloud import speech

environ["GOOGLE_APPLICATION_CREDENTIALS"] = "google-credentials.json"
speech_client = speech.SpeechClient()

with open("commands/check.mp3", 'rb') as f:
    byte_data = f.read()
audio = speech.RecognitionAudio(content=byte_data)


config = {
    "sample_rate_hertz": 16000,
    "language_code": "en-US",
    "speech_contexts": [{"phrases": ["p o w - r"], "boost":10.0}]
}

response = speech_client.recognize(config=config, audio=audio)

print("Transcript : ")
for result in response.results:
    print(result.alternatives[0].transcript)

File: server/check.js
const { readFileSync } = require("fs");
const { SpeechClient } = require("@google-cloud/speech");

process.env.GOOGLE_APPLICATION_CREDENTIALS = "google-credentials.json";

const client = new SpeechClient();

(async () => {
    const audio = {
        content: readFileSync("commands/test.mp3").toString("base64"),
    };
    console.log(audio.content.slice(0, 30));
    try {
        const [response] = await client.recognize({
            config: {
                sampleRateHertz: 48000,
                languageCode: "en-US",
                speechContexts: [
                    {
                        phrases: ["pow - r -"],
                        boost: 20.0,
                    },
                ],
            },
            audio,
        });
        const transcription = response?.results
            ?.map((result) => result.alternatives[0].transcript)
            .join("\n");
        console.log("Transcription: \n", transcription);
    } catch (error) {
        console.log(error);
    }
})();
